### ISSUE TITLE
Climate matching by drawing a buffer around occurrence points

### ISSUE THREAD

**Issue by soriadelva** [inbo/ClimateCastR#12]
At the moment, we keep points with every coordinate uncertainty level and also those that are close to the border with a different climate zone. To minimize bias related to this, draw a buffer around each occurrence point with the radius being equal to the coordinate uncertainty (i.e., records with a higher uncertainty will get a higher buffer around them). Then, calculate the amount of overlap between this buffer and the climate zone layer of the time period in which the observation was made. Based on this % overlap, calculate the presence of the point in each climate zone as follows: 60% overlap of occurrence record buffer with zone A and 40% with zone B =  0.6  occurrence records in zone A and 0.4 in zone B. 

In other words: n_climate (zone A) = Area of buffer in climate zone A / Total area of buffer (and this for all occurrence records of the species of interest)

**Comment by SanderDevisscher** [comment]
I would include some logic to, if a buffer overlaps also with the marine environment, keep only the terrestrial overlap, recalculate the area and the occurrence record buffer.
